File: apiwznm/WznmQMtyAMakefile.cpp
/**
	* \file WznmQMtyAMakefile.cpp
	* API code for table TblWznmQMtyAMakefile (implementation)
	* \author Alexander Wirthmueller
	* \date created: 25 Aug 2020
	* \date modified: 25 Aug 2020
	*/

#include "WznmQMtyAMakefile.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQMtyAMakefile
 ******************************************************************************/

WznmQMtyAMakefile::WznmQMtyAMakefile(
			const uint jnum
			, const string x1SrefKTag
			, const string titX1SrefKTag
			, const string Val
		) {
	this->jnum = jnum;
	this->x1SrefKTag = x1SrefKTag;
	this->titX1SrefKTag = titX1SrefKTag;
	this->Val = Val;
};

bool WznmQMtyAMakefile::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQMtyAMakefile");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "x1SrefKTag", "tag", x1SrefKTag);
		extractStringUclc(docctx, basexpath, "titX1SrefKTag", "tag2", titX1SrefKTag);
		extractStringUclc(docctx, basexpath, "Val", "val", Val);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQMtyAMakefile
 ******************************************************************************/

ListWznmQMtyAMakefile::ListWznmQMtyAMakefile() {
};

ListWznmQMtyAMakefile::ListWznmQMtyAMakefile(
			const ListWznmQMtyAMakefile& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQMtyAMakefile(*(src.nodes[i]));
};

ListWznmQMtyAMakefile& ListWznmQMtyAMakefile::operator=(
			const ListWznmQMtyAMakefile& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQMtyAMakefile(*(src.nodes[i]));

	return *this;
};

ListWznmQMtyAMakefile::~ListWznmQMtyAMakefile() {
	clear();
};

void ListWznmQMtyAMakefile::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQMtyAMakefile::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQMtyAMakefile* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQMtyAMakefile");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQMtyAMakefile", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQMtyAMakefile(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQMtyAMakefile[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwznm/WznmQMtyAMakefile.java
/**
  * \file WznmQMtyAMakefile.h
  * Java API code for record of table TblWznmQMtyAMakefile
  * \author Alexander Wirthmueller
  * \date created: 25 Aug 2020
  * \date modified: 25 Aug 2020
  */

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQMtyAMakefile {
	
	public WznmQMtyAMakefile(
				int jnum
				, String x1SrefKTag
				, String titX1SrefKTag
				, String Val
			) {
		this.jnum = jnum;
		this.x1SrefKTag = x1SrefKTag;
		this.titX1SrefKTag = titX1SrefKTag;
		this.Val = Val;
	};

	public int jnum;
	public String x1SrefKTag;
	public String titX1SrefKTag;
	public String Val;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQMtyAMakefile");

		if (Xmlio.checkXPath(doc, basexpath)) {
			x1SrefKTag = Xmlio.extractStringUclc(doc, basexpath, "x1SrefKTag", "tag", null, 0);
			titX1SrefKTag = Xmlio.extractStringUclc(doc, basexpath, "titX1SrefKTag", "tag2", null, 0);
			Val = Xmlio.extractStringUclc(doc, basexpath, "Val", "val", null, 0);
			
			return true;
		};

		return false;
	};
};

